api/data_test.go

```diff
@@ -65,6 +67,36 @@ func TestGetCacheTimeEndpoint(t *testing.T) {
 	})
 }
 
+func TestGetCachetimeReturnsError(t *testing.T) {
+	Convey("When a non-existant cache time is requested with its ID", t, func() {
+		ctx := context.Background()
+		r := httptest.NewRequest(http.MethodGet, baseURL+incorrectCacheID, http.NoBody)
+		w := httptest.NewRecorder()
+		mockedDataStore := &mock.DataStoreMock{
+			GetCacheTimeFunc: func(ctx context.Context, id string) (*models.CacheTime, error) {
+				return nil, errs.ErrCachetimeNotFound
+			},
+		}
+		dataStoreAPI := setupAPIWithStore(ctx, mockedDataStore)
+		dataStoreAPI.Router.ServeHTTP(w, r)
+		So(w.Code, ShouldEqual, http.StatusNotFound)
+	})
+	Convey("When the api cannot connect to datastore return an internal server error", t, func() {
+		ctx := context.Background()
+		r := httptest.NewRequest(http.MethodGet, baseURL+testCacheID, http.NoBody)
+		w := httptest.NewRecorder()
+		mockedDataStore := &mock.DataStoreMock{
+			GetCacheTimeFunc: func(ctx context.Context, id string) (*models.CacheTime, error) {
+				return nil, errs.ErrInternalServer
```

Comment:
This is not a realistic mock, as this is not how the actual function behaves. In fact, this `ErrInternalServer` error is only used here.

Reply:
I was unsure how write a meaningful test for Internal Errors, I kept my work in-line with the dp-dataset-api which also uses err.ErrInternalServer. I have added a line to make the use of this more explicit, but may need your assistence if the  ammendments still don't satisfy.

Reply:
address these issues